lib/system.ps1

```diff
@@ -0,0 +1,158 @@
+        [string]$Value,
+        [switch]$Global
+    )
+
+    $registerKey = if ($Global) {
+        Get-Item -Path 'HKLM:\SYSTEM\CurrentControlSet\Control\Session Manager'
+    } else {
+        Get-Item -Path 'HKCU:'
+    }
+    $envRegisterKey = $registerKey.OpenSubKey('Environment', $true)
+    if ($null -eq $Value -or $Value -eq '') {
+        $envRegisterKey.DeleteValue($Name)
+    } else {
+        $registryValueKind = if ($Value.Contains('%')) {
+            [Microsoft.Win32.RegistryValueKind]::ExpandString
+        } elseif ($envRegisterKey.GetValue($Name)) {
+            $envRegisterKey.GetValueKind($Name)
+        } else {
+            [Microsoft.Win32.RegistryValueKind]::String
+        }
+        $envRegisterKey.SetValue($Name, $Value, $registryValueKind)
+    }
+    Publish-EnvVar
+}
+
+function Find-Path {
+    param(
+        [string]$Name,
+        [string]$Path
+    )
+
+    if ($null -eq $Path -and $Path -eq '') {
+        return $false, $null
+    } else {
+        $strippedPath = $Path.Split(';', [System.StringSplitOptions]::RemoveEmptyEntries).Where({ $_ -ne $Name }) -join ';'
+        return ($strippedPath -ne $Path), $strippedPath
+    }
+}
+
+function Add-Path {
+    param(
+        [string]$Path,
+        [switch]$Global,
+        [switch]$Force
+    )
+
+    if (!$Path.Contains('%')) {
+        $Path = fullpath $Path
+    }
+    # future sessions
+    $inPath, $strippedPath = Find-Path $Path (Get-EnvVar -Name 'PATH' -Global:$Global)
+    if (!$inPath -or $Force) {
+        Write-Output "Adding $(friendly_path $Path) to $(if ($Global) {'global'} else {'your'}) path."
+        Set-EnvVar -Name 'PATH' -Value (@($Path, $strippedPath) -join ';') -Global:$Global
+    }
+    # current session
+    $inPath, $strippedPath = Find-Path $Path $env:PATH
+    if (!$inPath -or $Force) {
+        $env:PATH = @($Path, $strippedPath) -join ';'
+    }
+}
+
+function Remove-Path {
+    param(
+        [string]$Path,
+        [switch]$Global
+    )
+
+    if (!$Path.Contains('%')) {
+        $Path = fullpath $Path
+    }
+    # future sessions
+    $inPath, $strippedPath = Find-Path $Path (Get-EnvVar -Name 'PATH' -Global:$Global)
+    if ($inPath) {
+        Write-Output "Removing $(friendly_path $Path) from $(if ($Global) {'global'} else {'your'}) path."
+        Set-EnvVar -Name 'PATH' -Value $strippedPath -Global:$Global
+    }
+    # current session
+    $inPath, $strippedPath = Find-Path $Path $env:PATH
+    if ($inPath) {
+        $env:PATH = $strippedPath
+    }
+}
+
+## Deprecated functions
+
+function env($name, $global, $val) {
+    if ($PSBoundParameters.ContainsKey('val')) {
+        Show-DeprecatedWarning $MyInvocation 'Set-EnvVar'
+        Set-EnvVar -Name $name -Value $val -Global:$global
+    } else {
+        Show-DeprecatedWarning $MyInvocation 'Get-EnvVar'
+        Get-EnvVar -Name $name -Global:$global
+    }
+}
+
+function strip_path($orig_path, $dir) {
+    Show-DeprecatedWarning $MyInvocation 'Find-Path'
```

Comment:
I realize there are [references of this function in some manifests](https://github.com/search?q=org%3AScoopInstaller+strip_path+language%3AJSON&type=code), but I think this function should not be provided and used outside of the core _(though I know there is impossible to hide a function based on the current corebase, all functions are exposed, which I dislike)_.

What they really want is a way to manipulate `$env:PATH`-like environment variables, such as  `Add-PathLikeEnvVar` and `Remove-PathLikeEnvVar` _(further abstraction of `Add-Path` and `Remove-Path`)_. This is useful because there are many operations on `$env:PATH`-like env vars, plus the proposal of `$env:SCOOP_PATH` is staging. `Add-Path` and `Remove-Path` are `$env:PATH`-specific and may not be much useful if/after `$env:SCOOP_PATH` is implemented.

Reply:
I'll change function names ending with `-Path` to `-PathLikeEnvVar`. Maybe a refactoring into modules could be done if I have time.

For now, keep the deprecated warning of `strip_path`, and will remove outside use after this PR goes to `main`.